### PR TITLE
Modify rule S3433: Migrate to new educational format

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
Removing `swift` rule since it's not implemented yet.
I'll leave a comment in the rule as soon as this PR is merged.

### CODE REVIEW

File: rules/S3433/csharp/rule.adoc [hunk 1]
```diff
@@ -1,4 +1,61 @@
...
+[TestMethod]
+public void MyIgnoredTestMethod()
+{ /* ... */ }
+
+[TestMethod]
+public void MyIgnoredGenericTestMethod(int foo)
+{ /* ... */ }
+----
+
+=== Exceptions
```
Comment (sebastien-marichal) [inline rules/S3433/csharp/rule.adoc:46]: As per [documentation](https://github.com/SonarSource/rspec/blob/master/docs/description.adoc#2-education-format), the `Exceptions` should be under the `Why is this an issue?` section.

File: rules/S3433/csharp/rule.adoc [hunk 2]
```diff
@@ -1,4 +1,61 @@
...
+=== Compliant solution
+
+[source,csharp,diff-id=1,diff-type=compliant]
+----
+[TestMethod]
+public void TestNullArg()
+{  /* ... */  }
+
+[TestMethod]
+public void MyIgnoredTestMethod()
```
Comment (sebastien-marichal) [inline rules/S3433/csharp/rule.adoc:38]: Maybe not really important: an `async Task` compliant example could be nice.

File: rules/S3433/csharp/metadata.json [hunk 3]
```diff
@@ -1,3 +1,3 @@
 {
-  
+  "quickfix": "targeted"
```
Comment (sebastien-marichal) [inline rules/S3433/csharp/metadata.json:2]: Following my comment on the `async Task` test method, not sure if the quick fix is feasible as there are at least two possibilities in this case. 
If we are not required to cover all cases with the quick fix, you can ignore and resolve this comment.

**Review by sebastien-marichal — CHANGES_REQUESTED** [review]

File: rules/S3433/csharp/rule.adoc (hunk 1 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S3433/csharp/rule.adoc:46]: Yes, and it should be under the proper section, right? :worried:
Maybe what's confusing you is that we don't have a "How to fix it" section in this rule so the compliant/noncompliant examples are inside the "Why" section. Either that or I'm missing something, let me know

File: rules/S3433/csharp/rule.adoc (hunk 2 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S3433/csharp/rule.adoc:38]: Yeah that would be nice, my only concern is that having a 4/3 match on compliant/noncompliant examples can be confusing for users. WDYT?

File: rules/S3433/csharp/metadata.json (hunk 3 above)
Comment (cristian-ambrosini-sonarsource) [reply rules/S3433/csharp/metadata.json:2]: That's a good point, I'm assuming that `targeted` can result in either `partial` or `covered` once the quick fix is implemented so we should be fine.

File: rules/S3433/csharp/rule.adoc (hunk 2 above)
Comment (sebastien-marichal) [reply rules/S3433/csharp/rule.adoc:38]: I was thinking of replacing this compliant solution with the `async Task` one, keeping the 3/3 match.

**Review by sebastien-marichal — APPROVED** [review]
LGTM!

**Pull request merged**